src/Symfony/Component/Filesystem/Path.php

```diff
@@ -0,0 +1,990 @@
+            if ('' === $relativeBasePath) {
+                $relativePath = ltrim($relativePath, './\\');
+            }
+
+            return $relativePath;
+        }
+
+        // If the passed path is absolute, but the base path is not, we
+        // cannot generate a relative path
+        if ('' !== $root && '' === $baseRoot) {
+            throw new InvalidArgumentException(sprintf(
+                'The absolute path "%s" cannot be made relative to the '.
+                'relative path "%s". You should provide an absolute base '.
+                'path instead.',
+                $path,
+                $basePath
+            ));
+        }
+
+        // Fail if the roots of the two paths are different
+        if ($baseRoot && $root !== $baseRoot) {
+            throw new InvalidArgumentException(sprintf(
+                'The path "%s" cannot be made relative to "%s", because they '.
+                'have different roots ("%s" and "%s").',
+                $path,
+                $basePath,
+                $root,
+                $baseRoot
+            ));
+        }
+
+        if ('' === $relativeBasePath) {
+            return $relativePath;
+        }
+
+        // Build a "../../" prefix with as many "../" parts as necessary
+        $parts = explode('/', $relativePath);
+        $baseParts = explode('/', $relativeBasePath);
+        $dotDotPrefix = '';
+
+        // Once we found a non-matching part in the prefix, we need to add
+        // "../" parts for all remaining parts
+        $match = true;
+
+        foreach ($baseParts as $i => $basePart) {
+            if ($match && isset($parts[$i]) && $basePart === $parts[$i]) {
+                unset($parts[$i]);
+
+                continue;
+            }
+
+            $match = false;
+            $dotDotPrefix .= '../';
+        }
+
+        return rtrim($dotDotPrefix.implode('/', $parts), '/');
+    }
+
+    /**
+     * Returns whether the given path is on the local filesystem.
+     *
+     * @param string $path A path string.
+     *
+     * @return bool Returns true if the path is local, false for a URL.
+     *
+     * @since 1.0 Added method.
+     * @since 2.0 Method now fails if $path is not a string.
+     */
+    public static function isLocal(string $path): bool
+    {
+        return '' !== $path && false === strpos($path, '://');
+    }
+
+    /**
+     * Returns the longest common base path of a set of paths.
+     *
+     * Dot segments ("." and "..") are removed/collapsed and all slashes turned
+     * into forward slashes.
+     *
+     * ```php
+     * $basePath = Path::getLongestCommonBasePath(array(
+     *     '/webmozart/css/style.css',
+     *     '/webmozart/css/..'
+     * ));
+     * // => /webmozart
+     * ```
+     *
+     * The root is returned if no common base path can be found:
+     *
+     * ```php
+     * $basePath = Path::getLongestCommonBasePath(array(
+     *     '/webmozart/css/style.css',
+     *     '/puli/css/..'
```

Comment:
I suggest removing references to `puli` in the documents

Reply:
Replaced by  `symfony`